src/server/game/Config/GameConfig.cpp

```diff
@@ -0,0 +1,210 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "DatabaseEnv.h"
+#include "GameConfig.h"
+#include "Log.h"
+#include "Timer.h"
+#include "Util.h"
+
+GameConfig* GameConfig::instance()
+{
+    static GameConfig instance;
+    return &instance;
+}
+
+void GameConfig::CleanAll()
+{
+    _boolOptions.clear();
+    _intOptions.clear();
+    _floatOptions.clear();
+    _stringOptions.clear();
+}
+
+void GameConfig::AddBoolOption(std::string const& optionName, bool value /*= false*/)
+{
+    auto const& itr = _boolOptions.find(optionName);
+    if (itr != _boolOptions.end())
+    {
+        TC_LOG_FATAL("server.loading", "> Bool option (%s) is exist!", optionName.c_str());
```

Comment:
>Bool option (%s) is exist!

I think there is typo somewhere in this sentence, should it be "Bool option (%s) exists already!" or so ?

Reply:
Maybe, my english is bad :D